### ISSUE TITLE
Switch fully to a PR-based workflow and protect main branch

### ISSUE THREAD

**Issue by quamsta** [uiowa/brand-icon-browser#9]
From @pyrello 

> Hey. Can we switch to a PR-based workflow on https://github.com/uiowa/brand-icon-browser? The current situation of pushing updates to main is fine for single person development, but in a team it is much easier to have visibility into what is going on if there are PRs associated with all code changes. We should also set up main as a protected branch so that it is not possible to directly push to it.

I don't really have any strong objections to this. There were some far-reaching changes that I needed to make, so I merged my changes by myself from the `sharp` branch since `main` was falling too far behind. But I understand the benefit to making the process change, although I have no idea how to set up a protected branch, with all the fancy features that `uiowa` has. 

Any assistance would be greatly appreciated in setting this up in GitHub

**Comment by pyrello** [comment]
I set up the protected branch rules for `main` to match what we have for the uiowa/uiowa repo.

I believe that `main` is already set up as the default branch, so that means that when a new PR is opened, it will automatically be opened against that branch.

As for the workflow, here is roughly what we are doing on our other repos:
- When starting a new line of work, create a feature branch based on main:
  - `git checkout main && git pull`
  - `git checkout -b my_feature_branch`
- After you have some commits done, push the branch:
  - First time: `git push -u origin my_feature_branch`. If you forget, git prompts you with the correct command.
  - After: `git push`
- When pushing a new branch, you should receive a message back from GitHub with a link to open a new PR. Ideally a new PR should be opened ASAP as that will provide a cue to other developers that new work has started and the PR serves as a container for discussion related to the code being updated. If PR is still under active development, leave it in a draft state.
- Ideally before work is started there is also a corresponding issue that describes the work that is going to be done. If this is the case, you can use syntax like `Resolves #1` in the PR description to auto-close the issue when the PR is merged.
- When you are ready to have your code reviewed and the new feature tested, make sure your PR summary is up-to-date with testing instructions and then set it to "Ready for review".

**Comment by pyrello** [comment]
I enabled the following settings related to pull requests:
- Always suggest updating pull request branches - This provides the update button in the PR to merge in changes to `main`
- Automatically delete head branches  - This deletes the branch when the PR is merged or closed. Otherwise the branches have to be removed manually.

**Comment by pyrello** [comment]
@quamsta What else do we need to do related to this issue?

**Comment by quamsta** [comment]
Those settings are great, I was just thinking about the branches being deleted automatically. I'm not really sure what else is needed. This looks good for now. I'm okay with closing this out.

**Comment by quamsta** [comment]
I'm also going to remove some of the branches I've made and already merged into main, just to keep things tidy. Is there a preferred way to do that, @pyrello, or is it just as simple as deleting the branches?

**Comment by pyrello** [comment]
> I'm also going to remove some of the branches I've made and already merged into main, just to keep things tidy. Is there a preferred way to do that, @pyrello, or is it just as simple as deleting the branches?

I would just do it here: https://github.com/uiowa/brand-icon-browser/branches. No preferred way.

**Comment by pyrello** [comment]
Looks like this is all done, closing.
